packages/react/src/components/Code/index.tsx

```diff
@@ -18,30 +18,64 @@ const Highlighter = SyntaxHighlighter as React.ComponentType<
   SyntaxHighlighterProps
 >;
 
-interface Props extends SyntaxHighlighterProps {
+type Props = {
   children: string;
   language?: 'javascript' | 'css' | 'html' | 'yaml';
   className?: string;
-  tabIndex?: number;
-}
+  scrollable?: boolean;
+} & SyntaxHighlighterProps &
+  React.HTMLAttributes<HTMLDivElement>;
 
 const Code: React.ComponentType<React.PropsWithChildren<Props>> = ({
   children,
   className,
-  tabIndex,
+  scrollable = false,
   ...props
-}) => (
-  <>
+}) => {
+  const ref = useRef<HTMLPreElement>(null);
+  const [scrollableRegion, setScrollableRegion] = useState(false);
+  // react-syntax-highlighter does not provide direct access to its dom elements
+  // via refs, but we can specify the wrapping tags to bypass this limitation
+  // see: https://github.com/react-syntax-highlighter/react-syntax-highlighter/issues/335
+  const PreWithRef = (props: React.HTMLAttributes<HTMLPreElement>) => (
+    <pre {...props} ref={ref} />
+  );
+
+  useEffect(() => {
+    let observer: ResizeObserver;
+
+    // Track the containing element because resize observer will not
+    // trigger once an element becomes scrollable
+    if (scrollable && ref.current?.parentElement) {
+      const listener: ResizeObserverCallback = () => {
+        if (!ref.current) {
+          return;
+        }
+
+        const element = ref.current;
+        setScrollableRegion(element.clientWidth < element.scrollWidth);
+      };
+      const observer = new ResizeObserver(listener);
+      observer.observe(ref.current.parentElement);
+    }
+
+    return () => observer?.disconnect();
```

Comment:
```suggestion
    return () => {
      setScrollableRegion(false);
      observer?.disconnect();
    }
```

Reply:
Why is this needed? If the component is getting unmounted we don't need to update the state of the component.

Reply:
I assumed you had added `scrollable` and `ref.current` to the list of things for `useEffect` to watch for. I see now, though, that you left the list empty.  Is there a reason why you didn't add those to the list?

Reply:
> Ref objects don't always work as expected inside of useEffect because they're updated outside of the flow of a normal render, see comment https://github.com/facebook/react/issues/14387#issuecomment-503616820

You're right. I always forget about that.

> I hadn't considered there actually being a use case for allowing the scrollable prop to be changed after the initial render. Do you think that's a scenario that someone would use?

I think it's possible, such as if the layout changes to be responsive 🤷 But it's probably not a super common use case

Reply:
Sounds good. So should we apply this suggestion now?

Reply:
Whoops, look like I forgot to push that change 😅